### PR TITLE
RabbitMQ - Add username customization

### PR DESCRIPTION
**Pull request by MilenaHC**
Adds the ability to customize username generation for dynamic users in RabbitMQ.

Uses the new field `username_template` with the [go template language](https://golang.org/pkg/text/template/).

### CODE REVIEW

File: builtin/logical/rabbitmq/path_role_create_test.go [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+	username, exists := resp.Data["username"]
+	if !exists {
+		t.Fatalf("missing username in response")
+	}
+
+	require.Regexp(t, `^token-[a-z0-9]{8}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{4}-[a-z0-9]{12}$`, username)
+}
+
+func TestBackend_RoleCreate_CustomUsernameTemplate(t *testing.T) {
```
Comment (tomhjp) [inline builtin/logical/rabbitmq/path_role_create_test.go:87]: Also possibly similar enough for a table test? If you end up with lots of specialised logic based on the inputs, then it's not worth combining, but it looks like another good candidate to me.

**Review by tomhjp — COMMENTED** [review]
LGTM! 👍 Just some nits, and I think the API docs can get updated in this PR as well: `website/content/api-docs/secret/rabbitmq.mdx`.

File: builtin/logical/rabbitmq/path_config_connection.go [hunk 2]
```diff
@@ -65,6 +70,19 @@ func (b *backend) pathConnectionUpdate(ctx context.Context, req *logical.Request
 		return logical.ErrorResponse("missing password"), nil
 	}
 
+	usernameTemplate := data.Get("username_template").(string)
+	if usernameTemplate != "" {
+		up, err := template.NewTemplate(template.Template(usernameTemplate))
```
[the remark this answers is not in the document]
Comment (MilenaHC) [reply builtin/logical/rabbitmq/path_config_connection.go:75]: up = username producer

**Review by tomhjp — APPROVED** [review]
LGTM!

File: builtin/logical/rabbitmq/path_config_connection_test.go [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+		Password:         "password",
+		UsernameTemplate: "",
+	}
+
+	if !reflect.DeepEqual(actualConfig, expectedConfig) {
+		t.Fatalf("Expected: %#v\nActual: %#v", expectedConfig, actualConfig)
+	}
+}
+
+func TestBackend_ConfigConnection_CustomUsernameTemplate(t *testing.T) {
```
[the remark this answers is not in the document]
Comment (MilenaHC) [reply builtin/logical/rabbitmq/path_config_connection_test.go:58]: Talked with Tom about this on Slack and decided to leave the tests as they are.

**Review by jasonodonnell — APPROVED** [review]
LGTM :shipit: !

**Pull request merged**